### PR TITLE
[bazel] Upgrade workspace to Bazel 2.0.0.

### PR DESCRIPTION
**Pull request by jverkoey**
This change introduces the following improvements:

- Upgrade the workspace to support Bazel 2.0.0.
- Adds Travis CI tests for Bazel, xcodebuild, and pod lib lint.
- Standardizes the Bazel WORKSPACE, version, and flags configuration with Material iOS team conventions.
- Adds framework-style import support for Objective-C libraries that depend on this repo.
- Adds build tests.

In order to make `bazel test //...` work seamlessly from a local machine, this change unfortunately currently disables iOS 9 testing. This is because Catalina does not support iOS 9 testing and I was not able to find a way to conditionally include iOS 9 simulators as part of testing using `select` and Blaze platform constraints. See https://github.com/material-foundation/bazel-workspace-apple/issues/3 for the todo.

This change is part of the broader effort of https://github.com/material-components/material-components-ios/issues/9363

### CODE REVIEW

**Review by fluxminds — APPROVED** [review]

**Comment by jverkoey** [comment]
@wenyuzhang666 do you have ideas on how we might conditionally include the iOS 9 sim on platforms that support it?

**Comment by romoore** [comment]
Can any of these be broken-out into separate PRs? It seems like a laundry list in the PR description.

**Comment by jverkoey** [comment]
I’m not sure smaller PRs are worth the overhead in this case. Do you feel the added time to review + make the PRs is worth it?

**Comment by romoore** [comment]
I'd like to see some of those improvements land even if 2.0.0 has to be rolled-back for some reason.  However, you're the TL so you can ultimately decide.

File: BUILD [hunk 1]
```diff
@@ -99,7 +120,8 @@ ios_unit_test_suite(
     minimum_os_version = "9.0",
     timeout = "short",
     runners = [
-        ":IPAD_PRO_12_9_IN_9_3",
+# TODO(https://github.com/material-foundation/bazel-workspace-apple/issues/3): Conditionally enable iOS 9 tests on pre-Catalina devices.
```
Comment (romoore) [inline BUILD:123]: Can this be reverted and still upgrade to 2.0.0? Why does 2.0.0 require changes specifically for Catalina that earlier versions didn't require?

**Comment by jverkoey** [comment]
I'm willing to take on the risk of rollback for these changes because I expect to make a similar PR about ~10 times for 10 repos (https://github.com/material-components/material-components-ios/blob/162a063bdaff001a0a4aeba10c7ede5f0c96f3e0/WORKSPACE#L50-L108), and I'd like to avoid having to make this same family of changes 10 x ~5 times.

File: MDFTesting.podspec [hunk 2]
```diff
@@ -9,12 +9,12 @@ Pod::Spec.new do |s|
   s.platform     = :ios
   s.requires_arc = true
   s.ios.deployment_target = '9.0'
+  s.swift_version = '5.0'
```
Comment (romoore) [inline MDFTesting.podspec:12]: Is this a change from the default? I don't see it listed in the PR summary.

File: examples/MDFTestingExample/MDFTestingExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -187,7 +187,7 @@
 		DD871046233A7EFD0014F4B7 /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1100;
+				LastUpgradeCheck = 1120;
```
Comment (romoore) [inline examples/MDFTestingExample/MDFTestingExample.xcodeproj/project.pbxproj:190]: Are these changes required?  Do we need to lower the deployment target from 13 to 10 to upgrade to Bazel 2.0.0?

File: examples/Podfile.lock [hunk 4]
```diff
@@ -15,4 +15,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: f68cc4584957218d15c617848be26ebbd4bf5c8f
 
-COCOAPODS: 1.7.2
+COCOAPODS: 1.8.4
```
Comment (romoore) [inline examples/Podfile.lock:18]: Is this update required? I didn't see the CocoaPods version change listed in the PR summary.

**Review by romoore — COMMENTED** [review]
I'd like to request that this be split into separate PRs.

*   I don't know Travis `.yml` files well so I don't know that they're correct.
*   Several changes seem to have slipped-in without being mentioned in the PR summary. 
*   Many of these changes look unrelated.

**Comment by romoore** [comment]
Regarding your fear for the "10x5" amount of work, I'd be happy to help out writing the PRs for some of the "5".  If you know the things that have to be done, filing a "Do these N things in this project" type of Issue would allow me to help out and get it done faster.  What do you think?

**Comment by jverkoey** [comment]
Ah that's a healthy compromise! I'll write up the work required in a bit. Thanks Rob :)

**Comment by jverkoey** [comment]
I've updated https://github.com/material-components/material-components-ios/issues/9363 to include the full set of steps and work required to deploy bazel 2.0.0 across our fleet of dependencies. I'll close this PR and start tackling individual steps there instead.
